sdks/python/container/license_scripts/dep_urls_py.yaml

```diff
@@ -0,0 +1,116 @@
+  apache-beam:
+    license: "skip" # don't include self dependency.
+  appdirs:
+    license: "https://raw.githubusercontent.com/ActiveState/appdirs/master/LICENSE.txt"
+  avro:
+    license: "https://raw.githubusercontent.com/apache/avro/master/LICENSE.txt"
+    notice: "https://raw.githubusercontent.com/apache/avro/master/NOTICE.txt"
+  avro-python3:
+    license: "https://raw.githubusercontent.com/apache/avro/master/LICENSE.txt"
+    notice: "https://raw.githubusercontent.com/apache/avro/master/NOTICE.txt"
+  backports.weakref:
+    license: "https://raw.githubusercontent.com/PiDelport/backports.weakref/master/LICENSE"
+  chardet:
+    license: "https://raw.githubusercontent.com/chardet/chardet/master/LICENSE"
+  certifi:
+    license: "https://raw.githubusercontent.com/certifi/python-certifi/master/LICENSE"
+  cython:
+    license: "https://raw.githubusercontent.com/cython/cython/master/LICENSE.txt"
+  enum34:
+    license: "manual"
+  fastavro:
+    license: "https://raw.githubusercontent.com/fastavro/fastavro/master/LICENSE"
+    notice: "https://raw.githubusercontent.com/fastavro/fastavro/master/NOTICE.txt"
+  funcsigs:
+    license: "https://raw.githubusercontent.com/aliles/funcsigs/master/LICENSE"
+  futures:
+    license: "https://raw.githubusercontent.com/agronholm/pythonfutures/master/LICENSE"
+  google-apitools:
+    license: "https://raw.githubusercontent.com/google/apitools/master/LICENSE"
+  googledatastore:
+    license: "https://raw.githubusercontent.com/GoogleCloudPlatform/google-cloud-datastore/master/LICENSE"
+  grpcio:
+    license: "https://raw.githubusercontent.com/grpc/grpc/master/LICENSE"
+    notice: "https://raw.githubusercontent.com/grpc/grpc/master/NOTICE.txt"
+  grpcio-gcp:
+    license: "https://raw.githubusercontent.com/GoogleCloudPlatform/grpc-gcp-python/master/LICENSE"
+  guppy:
+    license: "https://raw.githubusercontent.com/joshwcomeau/guppy/master/LICENSE.md"
+  guppy3:
+    license: "https://raw.githubusercontent.com/zhuyifei1999/guppy3/master/LICENSE"
+  h5py:
+    license: "https://raw.githubusercontent.com/h5py/h5py/master/LICENSE"
+  hdfs:
+    license: "https://raw.githubusercontent.com/mtth/hdfs/master/LICENSE"
+  httplib2:
+    license: "https://raw.githubusercontent.com/httplib2/httplib2/master/LICENSE"
+  mock:
+    license: "https://raw.githubusercontent.com/testing-cabal/mock/master/LICENSE.txt"
+  monotonic:
+    license: "https://raw.githubusercontent.com/atdt/monotonic/master/LICENSE"
+  nose:
+    license: "https://raw.githubusercontent.com/nose-devs/nose2/master/license.txt"
+  numpy:
+    license: "https://raw.githubusercontent.com/numpy/numpy/master/LICENSE.txt"
+  oauth2client:
+    license: "https://raw.githubusercontent.com/googleapis/oauth2client/master/LICENSE"
+  pandas:
+    license: "https://raw.githubusercontent.com/pandas-dev/pandas/master/LICENSE"
+  pathlib2:
+    license: "https://raw.githubusercontent.com/mcmtroffaes/pathlib2/develop/LICENSE.rst"
+  protobuf:
```

Comment:
Alternative: we uninstall pip-licenses. Note that we need to do this in the same RUN command that installs it, otherwise it will not be deleted from the container. This way we can get rid of license_requirements.txt entriely.
```
COPY target/dep_urls_py.yaml target/pull_licenses_py.py /tmp/
RUN pip install -v /opt/apache/beam/tars/apache-beam.tar.gz[gcp] && \
   pip install pip-licenses && \  
   python /tmp/pull_licenses_py.py && \
   pip uninstall pip-licenses && \
   rm -rf /root/.cache/pip

# Log complete list of what exact packages and versions are installed.
RUN pip freeze --all
```

Reply:
I prefer the alternative method you suggested. Code changed to implement it.